MainPage.xaml.cs

```diff
@@ -22,81 +27,67 @@ public sealed partial class MainPage : Page
-
-            var headers = httpClient.DefaultRequestHeaders;
-            string header = "ie";
-            if (!headers.UserAgent.TryParseAdd(header))
+            string uri = "";
+            if (this.Keyword.Text.Length > 0)
             {
-                throw new Exception("Invalid header value: " + header);
+                uri = "https://api.github.com/search/repositories?q=" + this.Keyword.Text;
             }
-
-            header = "Mozilla/5.0 (compatible; MSIE 10.0; Windows NT 6.2; WOW64; Trident/6.0)";
-            if (!headers.UserAgent.TryParseAdd(header))
+            else
             {
-                throw new Exception("Invalid header value: " + header);
+                this.ResultList.ItemsSource = new List<Repo>();
+                this.ResultMsg.Text = "0 Result";
+                return;
             }
 
-            Uri requestUri = new Uri("https://api.github.com/search/repositories?q=" + this.Keyword.Text);
 
-            Windows.Web.Http.HttpResponseMessage httpResponse = new Windows.Web.Http.HttpResponseMessage();
-            string httpResponseBody = "";
+
+            string response = await HttpGetRequest(uri);
 
             try
             {
-                //Send the GET request
-                httpResponse = await httpClient.GetAsync(requestUri);
-                httpResponse.EnsureSuccessStatusCode();
-                httpResponseBody = await httpResponse.Content.ReadAsStringAsync();
 
-                Result result = Newtonsoft.Json.JsonConvert.DeserializeObject<Result>(httpResponseBody);
+                Result result = Newtonsoft.Json.JsonConvert.DeserializeObject<Result>(response);
+
+                Debug.WriteLine(result);
 
                 this.ResultMsg.Text = result.Items.Count() + " Results";
                 this.ResultList.ItemsSource = result.Items;
 
+                if (result.Items.Count().Equals(0))
+                {
+                    this.EmptyMsg.Visibility = Visibility.Visible;
+                    this.ResultList.Visibility = Visibility.Collapsed;
+                }
+                else if (!result.Items.Count().Equals(0) && result.Items.Count() < result.TotalCount)
+                {
+                    this.ResultList.ItemsSource = result.Items;
+                    this.ResultMsg.Text = "Top " + result.Items.Count() + " Results out of " + result.TotalCount + " Results";
+                }
+                else
+                {
+                    this.ResultList.ItemsSource = result.Items;
+                    this.ResultMsg.Text = result.Items.Count() + " Results";
+                }
             }
-            catch (Exception ex)
+            catch (Exception err)
             {
-                httpResponseBody = "Error: " + ex.HResult.ToString("X") + " Message: " + ex.Message;
-            }
+                ErrorResponse errorResponse = Newtonsoft.Json.JsonConvert.DeserializeObject<ErrorResponse>(response);
 
```

Comment:
@kimchhunnn This code is working ?

Reply:
@kenphanith It is didn't work well. I updated it.